internal/runners/builds/download.go

```diff
@@ -0,0 +1,175 @@
+package builds
+
+import (
+	"net/url"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/ActiveState/cli/internal/analytics"
+	"github.com/ActiveState/cli/internal/config"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/httputil"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/multilog"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/rtutils"
+	"github.com/ActiveState/cli/internal/runbits"
+	"github.com/ActiveState/cli/internal/runbits/rationalize"
+	"github.com/ActiveState/cli/internal/runbits/runtime"
+	auth "github.com/ActiveState/cli/pkg/platform/authentication"
+	"github.com/ActiveState/cli/pkg/platform/model"
+	"github.com/ActiveState/cli/pkg/platform/runtime/artifact"
+	"github.com/ActiveState/cli/pkg/platform/runtime/buildplan"
+	"github.com/ActiveState/cli/pkg/platform/runtime/setup/events"
+	"github.com/ActiveState/cli/pkg/platform/runtime/setup/events/progress"
+	"github.com/ActiveState/cli/pkg/platform/runtime/store"
+	"github.com/ActiveState/cli/pkg/platform/runtime/target"
+	"github.com/ActiveState/cli/pkg/project"
+	"github.com/go-openapi/strfmt"
+)
+
+type DownloadParams struct {
+	BuildID   string
+	OutputDir string
+}
+
+type Download struct {
+	out       output.Outputer
+	project   *project.Project
+	analytics analytics.Dispatcher
+	svcModel  *model.SvcModel
+	auth      *auth.Auth
+	config    *config.Instance
+}
+
+func NewDownload(prime primeable) *Download {
+	return &Download{
+		out:       prime.Output(),
+		project:   prime.Project(),
+		analytics: prime.Analytics(),
+		svcModel:  prime.SvcModel(),
+		auth:      prime.Auth(),
+		config:    prime.Config(),
+	}
+}
+
+func (d *Download) Run(params *DownloadParams) (rerr error) {
+	defer rationalizeError(&rerr)
+
+	if d.project == nil {
+		return rationalize.ErrNoProject
+	}
+
+	pg := runbits.NewRuntimeProgressIndicator(d.out)
+	defer rtutils.Closer(pg.Close, &rerr)
```

Comment:
This is for runtime progress, not for downloads of individual files. You should be able to use the progress library directly, we use "github.com/vbauerster/mpb/v7".

I would not expect to see any use of the runtime progress handler in this PR. That just so happens to be the only thing that uses "GetWithProgress", but that doesn't mean "GetWithProgress" is married to it.